### PR TITLE
[MultilevelMonteCarloApplication] Asynchronous algorithm and documentation

### PR DESCRIPTION
**Pull request by riccardotosi**
This PR introduces an asynchronous framework for both Monte Carlo and Multilevel Monte Carlo algorithms.
Basically this allows to have the machine always filled up, avoiding global synchronization points.
In addition, the application documentation is added.

### CODE REVIEW

**Comment by riccardotosi** [comment]
Ping @RiccardoRossi

File: applications/MultilevelMonteCarloApplication/python_scripts/adaptive_refinement_utilities.py [hunk 1]
```diff
@@ -15,65 +19,135 @@
...
-            # calculate the gradient of the variable
-            local_gradient = KratosMeshing.ComputeHessianSolMetricProcess(model_coarse.GetModelPart(model_part_name),local_gradient_variable,metric_param)
-            local_gradient.Execute()
-            # add again the removed variable parameter
-            metric_param.AddEmptyValue("local_gradient_variable")
-            metric_param["local_gradient_variable"].SetString(local_gradient_variable_string)
+            original_interp_error = metric_param["hessian_strategy_parameters"]["interpolation_error"].GetDouble()
+
+            # problem dependent section
+            if (problem_type == "body_fitted_ellipse"):
```
Comment (marcnunezc) [inline applications/MultilevelMonteCarloApplication/python_scripts/adaptive_refinement_utilities.py]: Here we are using `parameters["problem_data"]["problem_name"]` to distinguish between different cases, which is a user-defined name.

I would expect this to be something like "problem_type = potential_flow" or "problem_type = fluid_dynamics", something that is added in the parameters_refinement.

Comment (riccardotosi) [reply applications/MultilevelMonteCarloApplication/python_scripts/adaptive_refinement_utilities.py]: Done!

**Comment by riccardotosi** [comment]
@RiccardoRossi ping

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**